src/components/Widgets/Form/AddressSelect.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import shortid from 'shortid';
+import Identicon from '../../Identicon'
+import { combineWallets } from '../../../util/helpers';
+
+export default class AddressSelect extends Component {
+  static displayName = 'AddressSelect';
+
+  static propTypes = {
+    disabled: PropTypes.bool,
+    wallets: PropTypes.object,
+    walletContracts: PropTypes.object,
+    identiconSize: PropTypes.string,
+    onChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    disabled: false,
+    wallets: {},
+    walletContracts: {},
+    identiconSize: 'tiny',
+  };
+
+  state = {
+    selectedWallet: '',
+    combinedWallets: [],
+  }
+
+  componentDidMount() {
+    const { wallets, walletContracts, onChange } = this.props;
+    const combinedWallets = combineWallets(wallets, walletContracts);
+    const address = combinedWallets[0] ? combinedWallets[0].address : ''
+    this.setState(
+      { combinedWallets, selectedWallet: address },
+      onChange(combinedWallets[0])
+    )
+  }
+
+  selectAddress = e => {
+    const { onChange } = this.props;
+    this.setState({ selectedWallet: e.target.value }, onChange(e))
+  }
+
+  render() {
+    const { disabled, identiconSize } = this.props;
+    const { combinedWallets, selectedWallet } = this.state
+    return (
+      <StyledDiv>
+        <StyledSelect
+          onChange={this.selectAddress}
+          disabled={disabled}
+          value={selectedWallet}
+        >
+          {combinedWallets.map(w => {
+            return (
+              <StyledOption key={shortid.generate()} value={w.address}>
+                {
+                  w.addressType === 'wallet'
+                    ? `🔑 ${w.name}`
+                    : w['contract-name']
+                }
+                &nbsp; - &nbsp;
+                { w.balance }
+                &nbsp;
+                ETHER
+              </StyledOption>
+            );
+          })}
+        </StyledSelect>
+        <StyledIdenticon
+          title
+          size={identiconSize}
+          address={selectedWallet}
+        />
+        </StyledDiv>
+    );
+  }
+}
+
+const StyledOption = styled.option`
+  font-weight: normal;
+  display: block;
+  white-space: pre;
+  min-height: 1.2em;
+  padding: 0px 2px 1px;
+  color: #02a8f3;
+  line-height: 18px;
+  font-size: 1em;
+`
+
+const StyledSelect = styled.select`
```

Comment:
Did you try out the `react-select` component? Was there a reason you didn't use it?

Reply:
Just wanted to make sure to get in a PR on Monday, but I've just pushed a version using react-select.